core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts

```diff
@@ -245,8 +248,13 @@ export class TranslationSuggestionReviewModalComponent implements OnInit {
     this.isLastItem = this.remainingContributionIds.length === 0;
     this.allContributions = this.suggestionIdToContribution;
     this.allContributions[this.activeSuggestionId] = this.activeContribution;
-
     this.refreshActiveContributionState();
+    const tempDiv = this.renderer.createElement('div');
+    this.renderer.setProperty(tempDiv, 'innerHTML', this.translationHtml || '');
```

Comment:
Hey @StephenYu2018 , thanks for pointing out the security risk and for suggesting some alternative approaches.

Right now, I'm using innerHTML, but I see how that could pose a security risk. The linting rule (oppia/no-bypass-security-phrase) also means I can’t use bypassSecurityTrustHtml for sanitizing translationHtml.

I was considering regex to count the images, but I’m aware regex can be a bit unreliable for HTML parsing, especially with complex or malformed HTML, and it might miss edge cases like nested tags or different attribute variations.

Do you think regex could still work in this case, given the limitations, or would you recommend a better way to sanitize the HTML before I render and count the images?

Thanks !

Reply:
I'm not sure how exactly we can approach this. Initially, I was thinking about modifying by `textContent` instead of `innerHtml`. But I don't think we can add HTML elements within the `div` element's `textContent` property.

In regards to using regexes, I don't think we nest elements within non-interactive images. I think a regex like: `<oppia-noninteractive-image(?: *attribute-name="value")*><\/oppia-noninteractive-image>` with the global flag and the case insensitive flag could be used to get all the matches for images, and then we count each match as an image.

Reply:
I think a regex approach would be the way to go here as well, and I confirm that we don't nest elements within images. You can probably just check for the number of oppia-noninteractive-image tags along the lines that Stephen proposes.

Also, thanks, @StephenYu2018, for catching this issue!

Reply:
Also one question. How do we currently handle this "image count" check in the translation submission process? It would be good to standardize the approaches used there and here.

Reply:
@seanlip I believe the image count check is handled entirely on the backend (specifically in `/core/domain/suggestion_registry.py` with the `BaseSuggestion.get_new_image_filenames_added_in_suggestion()` method) in the translation submission process. But this information is needed on the frontend, and I'm having trouble finding any existing frontend functionality that counts HTML tags from strings.

Reply:
Ah ok, thanks @StephenYu2018! Yeah, in that case we'd need to build it, and I think the regex approach sounds reasonable. One suggestion is to maybe implement that as a separate "html parsing" service so that both the submission and review workflows can use it.